modules/logging-api-reference.adoc

```diff
@@ -0,0 +1,1155 @@
+|spec|object|  Specification of the desired behavior of ClusterLogForwarder
+|======================
+
+
+=== .spec
+===== Description
+ClusterLogForwarderSpec defines how logs should be forwarded to remote targets.
+
+=====  Type
+* object
+
+[options="header"]
+|======================
+|Property|Type|Description
+|inputs|array|  *(optional)* Inputs are named filters for log messages to be forwarded.
+|outputDefaults|object|  *(optional)* OutputDefaults are used to specify default values for OutputSpec
+|outputs|array|  *(optional)* Outputs are named destinations for log messages.
+|pipelines|array|  Pipelines forward the messages selected by a set of inputs to a set of outputs.
+|======================
+
+
+=== .spec.inputs[]
+===== Description
+InputSpec defines a selector of log messages.
+
+=====  Type
+* array
+
+[options="header"]
+|======================
+|Property|Type|Description
+|application|object|  *(optional)* Application, if present, enables named set of `application` logs that
+|name|string|  Name used to refer to the input of a `pipeline`.
+|======================
+
+
+=== .spec.inputs[].application
+===== Description
+Application log selector.
+All conditions in the selector must be satisfied (logical AND) to select logs.
+
+=====  Type
+* object
+
+[options="header"]
+|======================
+|Property|Type|Description
+|namespaces|array|  *(optional)* Namespaces from which to collect application logs.
+|selector|object|  *(optional)* Selector for logs from pods with matching labels.
+|======================
+
+
+=== .spec.inputs[].application.namespaces[]
+===== Description
+//Needs description
+
+
+=====  Type
+* array
+
+
+
+=== .spec.inputs[].application.selector
+===== Description
+A label selector is a label query over a set of resources.
+
+=====  Type
+* object
+
+[options="header"]
+|======================
+|Property|Type|Description
+|matchLabels|object|  *(optional)* matchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
+|======================
+
+
+=== .spec.inputs[].application.selector.matchLabels
+===== Description
+//Needs description
+=====  Type
+* object
+
+
+
+
+=== .spec.outputDefaults
+===== Description
+//Needs description
+=====  Type
+* object
+
+[options="header"]
+|======================
+|Property|Type|Description
+|elasticsearch|object|  *(optional)* Elasticsearch OutputSpec default values
+|======================
+
+
+=== .spec.outputDefaults.elasticsearch
+===== Description
+//Needs description
+=====  Type
+* object
+
+[options="header"]
+|======================
+|Property|Type|Description
+|enableStructuredContainerLogs|bool|  *(optional)* EnableStructuredContainerLogs enables multi-container structured logs to allow
+|structuredTypeKey|string|  *(optional)* StructuredTypeKey specifies the metadata key to be used as name of elasticsearch index
+|structuredTypeName|string|  *(optional)* StructuredTypeName specifies the name of elasticsearch schema
```

Comment:
> |enableStructuredContainerLogs|bool|  *(optional)* EnableStructuredContainerLogs enables multi-container structured logs to allow
|structuredTypeKey|string|  *(optional)* StructuredTypeKey specifies the metadata key to be used as name of elasticsearch index
|structuredTypeName|string|  *(optional)* StructuredTypeName specifies the name of elasticsearch schema

---
Truncated.